### DISCUSSION TITLE
Playing song info into text file

### DISCUSSION THREAD

**Discussion by richo67** [audacious-media-player/audacious#1512] [Q&A] [answered]
Hello,

I was wondering if there is any way in audacious (and its plugins) to write playing song's information into a text file?
What I mean is artist, song's title and possibly album to be written in the text file.

Thank you
-r-

**Comment by mschwendt** [comment]
You could quite easily base something on the "General > Song Change" plugin. Enable it, click on its tools icon to learn more about it.

**Reply by richo67** [reply]
Thank you for the tip. I was playing with it and trying to create some .bat file to run, but so far no success. I will try and will post something here once I succeed.

**Reply by mschwendt** [reply]
It seems to work even with a very simple one-liner directly within the plugin settings.

In the field "Command to run when starting a new song", I've added:

`echo "%a - %T - %b" >> ~/TEST`

Instead of echo you should be able to run your own shell script, command, whatever.

**Reply by richo67** [reply]
This works in linux, but unfortunately not in Windows.

**Reply by mschwendt** [reply, marked as answer]
> This works in linux, but unfortunately not in Windows.

```shell
cmd.exe /C echo "%a - %T - %b" >> C:\Users\USERNAME\test.txt
``` 

works here. I could think of running a batch file instead.

**Reply by richo67** [reply]
Michael,
thank you for all your effort and time. With your help I have managed :-) 

I am using now following string in "Command to tun when song title chnages" %n is string whcih one can define in playlist - Song Display. 
`cmd /c echo | set /p="%n" > C:\temp\current.music`

The part:  `| set /p=` is there to avoid extra line after the text

Also using following string when songs stops or end of playlist and end of song to empty the file
`cmd /c type nul >C:\temp\current.music`

I am using than this file to display it on background of my screen using bginfo tool from sysinternals. All works very well now.
Once again for your help!
-r-

**Reply by richo67** [reply]
There is one drawback of this approach:
Any active window loses focus for a fraction of a second at the time of these commands run. It is quite a problem when you are busy with drawing or other activities. your mouse or pen loses focus.

**Comment by mschwendt** [comment]
Well, my original suggestion was tested only with Linux. Executing an external program like Command Prompt on Windows causes side-effects possibly.

Given how unique your scenario and its requirements are, if it were on Linux where all development tools to compile Audacious are easily available, I would suggest modifying the available Song Change plugin to make it do directly what you want with system calls.
